script.js

```diff
@@ -0,0 +1,368 @@
+  setStyle(event.target, true);
+  identifyKey(event.target.dataset, 'down');
+  showKey();
+});
+
+keyboard.addEventListener('mouseup', function (event) {
+  textarea.focus();
+  if (event.target.dataset.code === 'CapsLock') return;
+  removeLastPressButtonStyle();
+  identifyKey(event.target.dataset, 'up');
+  showKey();
+});
+
+document.addEventListener('keydown', function (event) {
+  if (event.repeat) return;
+  event.preventDefault();
+  setStyle(document.querySelector(`.${keyboardElements[event.code][0]}`), true);
+  identifyKey(event, 'down');
+  checkLanguage();
+  showKey();
+});
+
+document.addEventListener('keyup', function (event) {
+  if (event.repeat) return;
+  if (event.key == 'CapsLock') return;
+  setStyle(
+    document.querySelector(`.${keyboardElements[event.code][0]}`),
+    false
+  );
+  identifyKey(event, 'up');
+  checkLanguage();
+  showKey();
+});
+
+function setStyle(code, bool) {
+  if (bool) code.classList.add('active');
+  else code.classList.remove('active');
+}
+
+function removeLastPressButtonStyle() {
+  lastPressButton.classList.remove('active');
+}
+
+function identifyKey(event, updown) {
+  if (event.code === 'ShiftLeft') {
+    if (shiftCaseLeft) shiftCaseLeft = false;
+    else shiftCaseLeft = true;
+  }
+
+  if (event.code === 'AltLeft') {
+    if (altCaseLeft) altCaseLeft = false;
+    else altCaseLeft = true;
+  }
+  if (event.code === 'CapsLock') {
+    if (capsLock === true) {
+      setStyle(
+        document.querySelector(`.${keyboardElements[event.code][0]}`),
+        false
+      );
+    }
+    capsButtonIsPress();
+  }
+
+  if (event.key === 'shift' || event.key === 'rShift' || event.key === 'Shift') {
+    capsButtonIsPress();
+    return;
+  }
+
+  if (updown === 'down') {
+    if (event.code === 'Tab') {
+      tabCase();
+    }
+
+    if (event.code === 'Backspace') {
+      backspaceButtonISPress();
+    }
+
+    if (event.code === 'Delete') {
+      deleteButtonIsPress();
+    }
+
+    if (shiftCase === false && sessionStorage.lang === 'ENG') {
+      if (keyboardElements[event.code][1] !== undefined) {
+        printKeyCode(keyboardElements[event.code][1]);
+      }
+    }
+
+    if (shiftCase === true && sessionStorage.lang === 'ENG') {
+      if (keyboardElements[event.code][2] !== undefined) {
+        printKeyCode(keyboardElements[event.code][2]);
+      }
+    }
+
+    if (shiftCase === false && sessionStorage.lang === 'RUS') {
+      if (keyboardElements[event.code][3] !== undefined) {
+        printKeyCode(keyboardElements[event.code][3]);
+      }
+    }
+
+    if (shiftCase === true && sessionStorage.lang === 'RUS') {
+      if (keyboardElements[event.code][4] !== undefined) {
+        printKeyCode(keyboardElements[event.code][4]);
+      }
+    }
+  }
+}
+
+function capsButtonIsPress() {
+  if (capsLock === true) capsLock = false;
+  else capsLock = true;
+
+  if (shiftCase === true) shiftCase = false;
+  else shiftCase = true;
+}
+
+function deleteButtonIsPress() {
```

Comment:
`deleteButtonIsPress`, `backspaceButtonISPress`, `printKeyCode` and `tabCase` are very similar. Can we remove duplication of the code?